design/pipeline-resources.md

```diff
@@ -27,73 +27,63 @@ resources:
 
 If you have a pipeline that produces artifacts, you can consume the artifacts by defining a `pipelines` resource. A pipeline can be another Azure DevOps pipeline or any external pipelines like Jenkins etc.
 
-
 ### Schema
 
 ```yaml
 resources:        # types: pipelines | repositories | containers | packages
   pipelines:
   - pipeline: string  # identifier for the pipeline resource
-    type: enum  # type of the pipeline source like azurePipelines, Jenkins etc. 
-    connection: string  # service connection to connect to the source
-    project: string # project for the source; optional for current project  
+    connection: string  # service connection for pipelines in other Azure DevOps organizations or Servers
+    project: string # project for the source; optional for current project
```

Comment:
We need to change `version` to `versionSpec`. When the plan is created, we need to resolve the pipeline resource to a specific version. "Latest" is not a "version". "Latest" is a version spec. It resolves to a specific build ID.

We need to stamp the resource with the resolved version, and store that in the generated plan environment.

All stages and jobs must resolve to the same version.

Reply:
And retried nodes should resolve to the same version (part of the same plan).

I would like to see retry/rerun behaviors called out explicitly. What happens during requeue?

Reply:
Leaving this feedback for @ashokirla

Reply:
The version here means the build number. By default, it will be resolved to latest successful pipeline run (across all stages).
If user wants to pick the latest from a specific branch or tag, user can specify the branch/tag inputs in the definition. Then we will fetch the latest successful pipeline from given branch/tag.

However, if user choses to consume a specific version, then he/she can give the pipeline run number as a 'version' input. In that case that version is picked. 

We will not consider any failed/cancelled pipeline runs while resolving the version. We will start with this simpler experience. Later if we see any valid scenarios for consuming artifacts from a failed pipeline, we can extend by introducing 'quality' and 'stage' as additional inputs as you have suggested.

And yes, we stamp the resource with the resolved version for any pipeline run instance. And the retry/redeploy scenarios also will use the same version. This is the same behavior we have in RM while resolving build artifacts. 

I am happy to get on a call if you have any concerns/feedback.